### PR TITLE
 Enable bitcode for iOS, allow modify deployment target for iOS/OSX

### PR DESCRIPTION
**Pull request by CAMOBAP795**
Tested on:
 - iOS:armv7
 - iOS:armv7s
 - iOS:armv8
 - iOS:i386
 - iOS:x86_64
 - OSX:x86_64

### CODE REVIEW

**Comment by CAMOBAP795** [comment]
@lasote I'm just starting to get acquainted with `conan` and common approaches to work with it. Could you please clarify:
 - How `testing/* -> release/*` transition happens, is there any serious testing before merge?
 - Should I modify travis/appveyor to setup cross compilation for iOS?

File: conanfile.py [hunk 1]
```diff
@@ -218,6 +218,11 @@ def find_cc(settings, sdk_name=None):
         if not str(self.settings.arch).startswith("arm"):
             cc += " -DOPENSSL_NO_ASM"
 
+        if "DARWIN_DEPLOYMENT_TARGET" in os.environ:
```
Comment (lasote) [inline conanfile.py]: From where it comes the variable "DARWIN_DEPLOYMENT_TARGET"?

Comment (lasote) [reply conanfile.py]: Probably you should use the `os.version` field for this purpose.

File: conanfile.py [hunk 2]
```diff
@@ -239,6 +244,9 @@ def osx_build(self, config_options_string):
         else:
             command = "./config %s %s" % (config_options_string, m32_suff)
 
+        if "DARWIN_DEPLOYMENT_TARGET" in os.environ:
```
Comment (lasote) [inline conanfile.py]: For macos we have an open issue, we are not distributing in the default `setting.yml` a subsetting `os.version` but we will in the next release: https://github.com/conan-io/conan/issues/2985
So I think we probably should wait and do it correctly.

**Comment by lasote** [comment]
Hi! 

> How testing/* -> release/* transition happens, is there any serious testing before merge?

The transition is done by me, once it is merged.

> Should I modify travis/appveyor to setup cross compilation for iOS?

Would be great if possible, only travis.

File: conanfile.py (hunk 1 above)
Comment (CAMOBAP795) [reply conanfile.py]: > From where it comes the variable "DARWIN_DEPLOYMENT_TARGET"?

Kind of user defined

> Probably you should use the os.version field for this purpose.

Wow! Looks like it's perfectly fit for this, thanks for advice, I will update PR

Comment (CAMOBAP795) [reply conanfile.py]: @lasote done

File: conanfile.py (hunk 2 above)
Comment (CAMOBAP795) [reply conanfile.py]: Agree, done

**Comment by lasote** [comment]
Looks good! let's wait for the CI. Ping me when it finishes if I do not write first.

**Comment by helmesjo** [comment]
@lasote It finished! :)

**Pull request merged**